server/consensus/consensus.go

```diff
@@ -203,8 +262,10 @@ func MainConsensus(recvc chan BackendMessage, sendc chan ConsensusMessage, serve
 	}
 	if err := nh.StartOnDiskCluster(peers, join, stateMachineProvider, rc); err != nil {
 		fmt.Fprintf(os.Stderr, "failed to add cluster, %v\n", err)
-		os.Exit(1)
 	}
+
+	go ScanDiscoveryService(servers, nh)
```

Comment:
do all the nodes try to discover other nodes? or is it just one of them?

Reply:
All the nodes discover other nodes. This way there is not one point of failure for nodes joining the cluster.